ideahacks/routes/admin/parts.js

```diff
@@ -15,7 +37,10 @@ const postParts = (req, res) => {
   })
   newPart.save()
 
-  res.json({ message: 'Received POST request' })
+  res.json({ message: 'Success! Part has been created' })
+  //end of part creation
```

Comment:
After the uniqueness check, everything is good to go, so we create a new part, save it into the database, and then respond with a success message.

1. With the success JSON, can you please add `status: 'success'` into the JSON.

2. Can you please move the success steps **into** the `.then()` function of the uniqueness test?
Example:
```
Part.find(.....).then(parts => {
  // if not unique, respond with status failure and an error message

  // create new part here
  // save new part here
  // response with status success and success message here
}
```

The reason we do this is because queries into our database with the `.find()` function is considered **asynchronous I/O**. Basically, when we use the `.find()` function to search our database for an already existing part, the function will essentially say "hey, I'm going to brb as I search our db. Go ahead and execute all the other lines of javascript until I finish." So actually all the lines of code under our `Part.find()` function will run first, which is where our success logic is found. It will create, save, and return success. Finally, the `Part.find()` query will complete and be like "hey, I'm back and found what you wanted" and will then execute the `.then()` function. But we run into a problem here, because the function will try to return status failure, but something has already been returned (by running the success steps prior). This causes a warning.

We avoid this issue by moving the success step into the `.then()` function, so that when the `Part.find()` query executes, there'll be nothing under it to execute, and thus we will be forced to wait for it to return and fire off the `.then()` function. Only when the uniqueness test passes will we run through the success steps, thus solving our problem, and guarantee'ing that we'll return only **ONE** JSON per request.